Enhancing docs about PRNG seeding (#7068)
numba/numba: docs/source/reference/numpysupported.rst

* :func:`numpy.random.seed`: with an integer argument only

.. warning::

>    Calling :func:`numpy.random.seed` from non-Numba code (or from :term:`object mode`
>    code) will seed the NumPy random generator, not the Numba random generator.

Reviewer: ```suggestion
   Calling :func:`numpy.random.seed` from interpreted code (including from :term:`object mode`
   code) will seed the NumPy random generator, not the Numba random generator.
```

>    To properly seed the Numba random generator, see example below.

Reviewer: ```suggestion
   To seed the Numba random generator, see the example below.
```

.. code-block:: python

  from numba import njit
  import numpy as np

  @njit
  def seed(a):
      np.random.seed(a)

  @njit
  def rand():
      return np.random.rand()

>   # Uncorrect seeding

Reviewer: ```suggestion
  # Incorrect seeding
```
